Remove legacy build system script phases and build settings

emw-apple opened the pull request:
#### c530af3288e16ca55f3dad0ee69f2e5a1d90990b
<pre>
Remove legacy build system script phases and build settings
<a href="https://bugs.webkit.org/show_bug.cgi?id=243221">https://bugs.webkit.org/show_bug.cgi?id=243221</a>
rdar://89701972

Reviewed by Alexey Proskuryakov.

Delete WK_WHICH_BUILD_SYSTEM and WK_USE_NEW_BUILD_SYSTEM settings.
Remove script phases which checked these settings and do nothing under
XCBuild.

* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/Configurations/ToolExecutable.xcconfig:

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj: Add
  new target dependencies from JavaScriptCore to the LLInt targets, now
  that the legacy-only aggregate targets are gone. Avoid a spurious
  dependency cycle in non-workspace builds by setting
  BuildIndependentTargetsInParallel (this was the cause of an earlier
  revert).

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libabsl.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/WGSL.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/DumpRenderTreeLibrary.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/LayoutTestHelper.xcconfig:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWTFLibrary.xcconfig:
</pre>

emw-apple commented on Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
```diff
@@ -4538,10 +4538,11 @@
 			filePatterns = asm;
 			fileType = sourcecode.asm.asm;
 			inputFiles = (
+				"$(BUILT_PRODUCTS_DIR)/yasm",
```
This input is added because a script phase I removed below told me to:

```
# Once support for the legacy build system is dropped, delete this build phase
# and move its input file to the "Assembly files" build rule.
```

aproskuryakov reviewed (approved):
I'm a little surprised that this is all xcconfig changes, and nothing in `Tools/Scripts`.

emw-apple committed:
Remove legacy build system script phases and build settings
https://bugs.webkit.org/show_bug.cgi?id=243221
rdar://89701972

Reviewed by Alexey Proskuryakov.

Delete WK_WHICH_BUILD_SYSTEM and WK_USE_NEW_BUILD_SYSTEM settings.
Remove script phases which checked these settings and do nothing under
XCBuild.

* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/Configurations/ToolExecutable.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/ANGLE-dynamic.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libabsl.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/WGSL.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/DumpRenderTreeLibrary.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/LayoutTestHelper.xcconfig:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWTFLibrary.xcconfig:

Canonical link: https://commits.webkit.org/252870@main
